bellatrix.plugins.visual-regression-tracker/src/main/java/solutions/bellatrix/plugins/vrt/VisualRegressionService.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package solutions.bellatrix.plugins.vrt;
+
+import io.visual_regression_tracker.sdk_java.TestRunOptions;
+import io.visual_regression_tracker.sdk_java.TestRunResult;
+import io.visual_regression_tracker.sdk_java.VisualRegressionTracker;
+import io.visual_regression_tracker.sdk_java.VisualRegressionTrackerConfig;
+import plugins.screenshots.ScreenshotPlugin;
+import solutions.bellatrix.core.configuration.ConfigurationService;
+import solutions.bellatrix.core.utilities.SingletonFactory;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class VisualRegressionService {
+    private static float defaultDiffTolerance() {
+        return ConfigurationService.get(VisualRegressionSettings.class).getDefaultDiffTolerance();
+    }
+
+    private static final ThreadLocal<VisualRegressionTracker> VISUAL_REGRESSION_TRACKER_THREAD_LOCAL;
+    private static final ThreadLocal<VisualRegression> VISUAL_REGRESSION_ATTRIBUTE;
+
+    static {
+        try {
+            VISUAL_REGRESSION_TRACKER_THREAD_LOCAL = new ThreadLocal<>();
+            VISUAL_REGRESSION_ATTRIBUTE = new ThreadLocal<>();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to initialize VisualRegressionService", e);
+        }
+    }
+
+    public static VisualRegressionTracker tracker() {
+        return VISUAL_REGRESSION_TRACKER_THREAD_LOCAL.get();
+    }
+
+    public static void initService(VisualRegression attribute, VisualRegressionSettings settings) {
+        VISUAL_REGRESSION_ATTRIBUTE.set(attribute);
+
+        var config = new VisualRegressionTrackerConfig(
+                settings.getApiUrl(),
+                settings.getApiKey(),
+                !attribute.projectName().isBlank() ? attribute.projectName() : settings.getProject(),
+                settings.getBranch(),
+                settings.getCiBuildId(),
+                settings.isEnableSoftAssert(),
+                settings.getHttpTimeout()
+        );
+
+        VISUAL_REGRESSION_TRACKER_THREAD_LOCAL.set(new VisualRegressionTracker(config));
+    }
+
+    public static TestRunResult track(String name, Float diffTolerance, String screenshot) {
+        try {
+            return tracker().track(name, screenshot, buildRunOptions(diffTolerance));
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static TestRunResult track(String name, Float diffTolerance) {
+        return track(name, diffTolerance, takeSnapshot(name));
+    }
+
+    public static TestRunResult track(String name) {
+        return track(name, defaultDiffTolerance());
+    }
+
+    public static String takeSnapshot(String name) {
+        var screenshotPlugin = SingletonFactory.getInstance(ScreenshotPlugin.class);
+        return Objects.requireNonNull(screenshotPlugin).takeScreenshot(name);
```

Comment:
In my opinion, it would be beneficial to provide a more descriptive exception message to facilitate easier debugging.  For example, ScreenshotPlugin is not registered. or something like that.